client/settings/setting-option.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import {
+	CheckboxControl,
+	SelectControl,
+	TextareaControl,
+	TextControl,
+} from '@wordpress/components';
+import { NumberControl } from '@woocommerce/experimental';
+
+export const SettingOption = ( {
+	description,
+	label,
+	onChange = () => {},
+	options,
+	placeholder,
+	type,
+	value,
+} ) => {
+	if ( ! type ) {
+		return null;
+	}
+
+	const renderControl = () => {
```

Comment:
I don't know how many more field types we might get, and or customizations that can happen when rendering a setting component
If so, what I have done in the past is to create some kind of registry (look up table), where you register the field types with conditions. Either just `type` or more if needed.
It would return the component, and every component would take the same props. It could allow 3PDs to register their own customer setting definition, and use it to render something.

Kinda like this (doing it on the fly, this is very basic):
```
const SettingComp = SettingRegistry.get(type);

return <SettingComp 
  value={ value }
  onChange={ onChange }
  placeholder={ placeholder }
  type={ type }
   />;
```
Registry:
```
const text = {
  type: 'text',
  component: TextControl // or a custom one
  // anything else that could be of relevance.
};
SettingRegistry.register(text);
```

Reply:
This is a good idea, but this is all of the types with the exception of HTML sections.  We will allow custom components, but plan to do so using `SlotFill`.

Reply:
Nice, then my above suggestion would be overkill.
Just didn't want to end up at a point where we have a bunch of different setting types and use cases that could make a mess, and could cause for a lot of bugs.